addons/website_event/tests/test_event_internals.py

```diff
@@ -206,3 +214,45 @@ def test_registration_answer_search(self):
         # should fetch "registration_2" because the answer to the first question is "Q1-Answer2"
         # should fetch "registration_3" because the answer to the third question is "Answer2" (as free text)
         self.assertEqual(search_res, registration_2 | registration_3)
+
+    @users('portal', 'admin', 'user_public')
+    def test_user_can_see_on_website(self):
+        """
+        Tests the computed field `can_see_on_website` on `event.event`.
+        `filtered_domain` is used to check if the `_compute_can_see_on_website` method is correct
+        `search` is used to check if the `_search_can_see_on_website` method is correct
+        """
+
+        event_everyone, event_link, event_portal = [
+            self.env['event.event'].sudo().create(
+                {'name': 'event', 'visibility': visibility}
+            ) for visibility in ['everyone', 'link', 'portal_users']
+        ]
+        events_rset = event_everyone | event_link | event_portal
+
+        events_user_can_see_compute = events_rset.filtered_domain([('can_see_on_website', '=', True)])
+        events_user_can_see_search = events_rset.search([
+            ('can_see_on_website', '=', True),
+            ('id', 'in', events_rset.ids),
+        ])
```

Comment:
to avoid doing this in each of the tests, we could have a small method that do this job and return a dict with the info we need for the tests.